### DISCUSSION TITLE
Using FakePlayer with Entity AI

### DISCUSSION THREAD

**Discussion by TropicalShadow** [Minestom/Minestom#1642] [Help] [answered]
Can FakePlayers be used inside the AI system? if not is there any way of programming AI into the Fake Player using pre-existing code in the AI system?

**Comment by iam4722202468** [comment, marked as answer]
Depending on your use case, fake player may not be the correct way to do this. FakePlayer is a class that lets you create a player, which behaves identically to normal players. For an NPC or mob, you should extend entity and spawn an entity of type player.

With some small changes you should be able to use something like [this](https://gist.github.com/iam4722202468/36630043ca89e786bb6318e296f822f8), which will let you use AI
